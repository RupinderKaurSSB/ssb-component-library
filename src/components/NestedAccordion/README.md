NestedAccordion
========

> Nested accordion

### Usage

#### HTML

```html
<div class="ssb-nested-accordion mt-3">
  <div class="nested-accordion-header closed" onclick="{toggle classname to 'open'}">
    {plus icon}
    <span class="header-text">Title</span>
  </div>
  <div class="accordion-body closed">
    {insert content}
  </div>
</div>
```
Toggle the all instances of the `closed` className to `open` when clicked.

#### React

```jsx harmony
<Accordion primary header="This is a primary header">
  This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
  <NestedAccordion header="A nested accordion header">
    This is paragraph text which explains the accordion, the rest of the text is just to fill it out and show the space it takes.
  </NestedAccordion>
</Accordion>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children | node | Accordion content |
| header | string | Header text |
| openByDefault | bool | Will set the open state on init |

