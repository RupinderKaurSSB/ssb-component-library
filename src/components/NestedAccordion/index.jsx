import React, { useState } from 'react';
import PropTypes from 'prop-types';

const NestedAccordion = ({
	children, header, openByDefault,
}) => {
	const [isOpen, toggleOpen] = useState(openByDefault);
	return (
		<div className="ssb-nested-accordion mt-3">
			<button
				className={`nested-accordion-header ${isOpen ? 'open' : 'closed'}`}
				onClick={() => toggleOpen(!isOpen)}
			>
				<span className="button-grid">
					{!isOpen && <i className="expand-icon feather-chevron-down" />}
					{isOpen && <i className="expand-icon feather-chevron-up" />}
					<span className="header-text">{header}</span>
				</span>
			</button>
			<div className={`nested-accordion-body ${isOpen ? 'open' : 'closed'}`}>
				{isOpen && children}
			</div>
		</div>
	);
};

NestedAccordion.defaultProps = {
	openByDefault: false,
};

NestedAccordion.propTypes = {
	children: PropTypes.node,
	header: PropTypes.string,
	openByDefault: PropTypes.bool,
};

export default NestedAccordion;
