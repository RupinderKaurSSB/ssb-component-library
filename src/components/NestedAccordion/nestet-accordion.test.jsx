import React from 'react';
import {shallow} from 'enzyme';
import NestedAccordion from './index';

describe('NestedAccordion component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<NestedAccordion header="A nested accordion header">Nestet accordion</NestedAccordion>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Change classname to open on click', () => {
		const wrapper = shallow(<NestedAccordion header="A nested accordion header">Nestet accordion</NestedAccordion>);
		wrapper.find('.nested-accordion-header').simulate('click');
		expect(wrapper.find('.nested-accordion-header').hasClass('open')).toEqual(true);
		expect(wrapper.find('.nested-accordion-body').hasClass('open')).toEqual(true);
	});

});
