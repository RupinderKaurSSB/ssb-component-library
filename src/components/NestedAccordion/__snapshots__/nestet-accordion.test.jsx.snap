// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedAccordion component Matches the snapshot 1`] = `
<div
  className="ssb-nested-accordion mt-3"
>
  <button
    className="nested-accordion-header closed"
    onClick={[Function]}
  >
    <span
      className="button-grid"
    >
      <i
        className="expand-icon feather-chevron-down"
      />
      <span
        className="header-text"
      >
        A nested accordion header
      </span>
    </span>
  </button>
  <div
    className="nested-accordion-body closed"
  />
</div>
`;
